interfaces/cython/cantera/composite.py

```diff
@@ -649,6 +649,13 @@ def append(self, state=None, **kwargs):
           the full-state setters::
 
               mystates.append(T=300, P=101325, X={'O2':1.0, 'N2':3.76})
+
+        By default, the mass or mole fractions will be normalized so they
+        sum to 1.0. If this is not desired, the ``normalize`` argument can
+        be set to ``False``. In this case, the mass or mole fractions must
+        be specified as an array. For example::
```

Comment:
Per discussions elsewhere, I believe the intuitive approach is to not normalize when appending a state or a numpy array specifying X or Y. If the composition is provided using a composition, then normalization should be the default.

I am suggesting this, as the state holds the 'true' information (as `Y`), and thus should *not* be changed. Further, a user presumably expects setters for numpy arrays to remain unchanged (even if `X` cannot be retrieved without being normalized for the output).

I am mainly commenting on the docstring here, and haven't checked the code, as I am starting to lose track on the overarching philosophy. Could you prepare a post (regular comment, not a reply here) summarizing the overall strategy when to normalize and when not to? At this point it may be good to see this documented in a single point.

Reply:
I think that would be a pretty substantial inconsistency with the behavior of the C++ `Phase` class, which has always normalized mass and mole fractions, even when they are provided as arrays.